java/ql/src/semmle/code/configfiles/ConfigFiles.qll

```diff
@@ -0,0 +1,101 @@
+/**
+ * Provides classes and predicates for working with configuration files, such
+ * as Java .properties or .ini files.
+ */
+
+import semmle.code.Location
+
+/** Each element in a configuration file has a location. */
+abstract class ConfigLocatable extends @configLocatable {
+  /** Retrieve the source location for this element. */
+  Location getLocation() {
+    configLocations(this, result)
+  }
+
+  /** The file associated with this element. */
+  File getFile() {
+    result = getLocation().getFile()
+  }
+
+  /** Gets a textual representation of this element. */
+  abstract string toString();
+}
+
+/**
+ * A name-value pair often used to store configuration properties
+ * for applications, such as the port, name or address of a database.
+ */
+class ConfigPair extends @config, ConfigLocatable {
+
+  /** If this ConfigPair has a name element, return it. */
```

Comment:
Please make sure that the QLDoc adheres to the [style guide](https://github.com/Semmle/ql/blob/master/docs/ql-style-guide.md#documentation), both here and elsewhere in this file.

For example, the QLDoc above could be written as: "Gets the name component of this `ConfigPair`, if any."

Reply:
Also, be sure to run this through the autoformatter.  That's ctrl+shift+F in eclipse.